packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -0,0 +1,123 @@
+import SectionHead from '@celo/react-components/components/SectionHead'
+import React, { useMemo, useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { SectionList, View } from 'react-native'
+import { useDispatch } from 'react-redux'
+import config from 'src/geth/networkConfig'
+import useInterval from 'src/hooks/useInterval'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import useSelector from 'src/redux/useSelector'
+import { tokensByAddressSelector } from 'src/tokens/selectors'
+import { updateTransactions } from 'src/transactions/actions'
+import { TRANSACTIONS_QUERY } from 'src/transactions/feed/query'
+import TransferFeedItem from 'src/transactions/feed/TransferFeedItem'
+import NoActivity from 'src/transactions/NoActivity'
+import { transactionsSelector } from 'src/transactions/reducer'
+import { FeedType } from 'src/transactions/TransactionFeed'
+import { TokenTransaction } from 'src/transactions/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import Logger from 'src/utils/Logger'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+const TAG = 'transactions/TransactionFeed'
+// Query poll interval
+export const POLL_INTERVAL = 10000 // 10 secs
+
+interface TransactionFeed {
+  tokenTransactionsV2: {
+    __typename: 'TokenTransactionsV2'
+    transactions: TokenTransaction[]
+  }
+}
+
+function useQueryTransactionFeed() {
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const dispatch = useDispatch()
+
+  // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
+  const [counter, setCounter] = useState(0)
+  useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
+
+  // TODO: Extract this to a more generic function/hook so that it can be reused
+  const { loading, error, result } = useAsync(
+    async () => {
+      const response = await fetch(`${config.blockchainApiUrl}/graphql`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+        body: JSON.stringify({
+          query: TRANSACTIONS_QUERY,
+          variables: { address, localCurrencyCode },
+        }),
+      })
+      return response.json()
+    },
+    [counter],
+    {
+      onSuccess: (result) => {
+        if (result?.data?.tokenTransactionsV2.transactions.length) {
+          dispatch(updateTransactions(result.data.tokenTransactionsV2.transactions))
+        }
+      },
+    }
+  )
+
+  if (result?.errors) {
+    Logger.warn(
+      TAG,
+      `Found errors when querying the transaction feed: ${JSON.stringify(result.errors)}`
+    )
+  }
```

Comment:
nit: Could be nice to put this one in the `useAsync` `onError` option.

Reply:
The problem is that this isn't an error, GraphQL returns 200 for errors as well, but I can put it in `onSuccess`

Reply:
oh yep! 👍